tiledb/core.cc

```diff
@@ -1361,14 +1370,18 @@ class PyQuery {
   void submit_write() {}
 
   void submit() {
-    if (array_->query_type() == TILEDB_READ)
+    if (array_->query_type() == TILEDB_READ) {
       submit_read();
-    else if (array_->query_type() == TILEDB_WRITE)
+    } else if (array_->query_type() == TILEDB_WRITE) {
       submit_write();
-    else if (array_->query_type() == TILEDB_DELETE)
+    } else if (array_->query_type() == TILEDB_DELETE) {
       query_->submit();
-    else
+      if (query_->query_status() == Query::Status::UNINITIALIZED) {
```

Comment:
Is this placed only in the `else if (array_->query_type() == TILEDB_DELETE)` block on purpose?

Reply:
The other branches don’t call submit directly right here